pydantic/main.py

```diff
@@ -167,6 +168,9 @@ class BaseModel(metaclass=_model_construction.ModelMetaclass):
     __pydantic_validator__: ClassVar[SchemaValidator | PluggableSchemaValidator]
     """The `pydantic-core` `SchemaValidator` used to validate instances of the model."""
 
+    __pydantic_validate_call_infos__: ClassVar[Dict[str, ValidateCallInfo]] = {}  # noqa: UP006
+    """Metadata containing functions decorated with `validate_call`; not inherited by subclasses."""
```

Comment:
If this is not inherited by subclasses, does that mean the following would misbehave:

```python
class A[T](BaseModel):
    @validate_call(validate_return=True)
    def f(self, x: T) -> T:
        return x

class B[T](A[T]):
    pass

b = B[int]()
b.f('abc')  # throw error correctly
```
I would think the third line should error, I didn't see a test that looked to cover this case in your tests below, maybe I missed it...

Reply:
Nevermind it does look like this is tested.